mxresolv/mxresolv.go

```diff
@@ -146,13 +155,14 @@ func isASCII(s string) bool {
 }
 
 type lookupResult struct {
-	mxHosts  []string
-	implicit bool
-	err      error
+	mxRecords []*net.MX
+	mxHosts   []string
+	implicit  bool
+	err       error
 }
 
-func cacheAndReturn(hostname string, mxHosts []string, implicit bool, err error) (retMxHosts []string, retImplicit bool, reterr error) {
-	lookupResultCache.AddWithTTL(hostname, lookupResult{mxHosts, implicit, err}, cacheTTL)
+func cacheAndReturn(hostname string, mxHosts []string, mxRecords []*net.MX, implicit bool, err error) (retMxHosts []string, retImplicit bool, reterr error) {
```

Comment:
I would remove mxHosts from cache at all, they are not needed because we rely on mxRecords anyway.

Reply:
I tried that first, but then noticed we cache Lookup errors by returning the recipient domain. So I ended up with this implementation.

If we don't want to cache mxHosts on errors, then I can do that.

Reply:
I see, it is needed, for the case of using A record if MX records are not provided. But we can skip storing mxHosts for the case when mxRecords are available to save space.